[tellstick] Sensor item updates get duplicated close to 100 times after lowering refresh interval

## Expected Behavior
Lowering the refresh interval simply updates states of lights faster.

## Current Behavior
The state of lights is updated faster as expected, but in addition, the sensor updates are repeated close to 100 times within less than a second.

Notes about the behavior:
- The default refresh interval was changed from 60 s to 3 s
- After resetting the refresh interval back to the original, issue disappeared
- Issue is accumulating over time. Within an hour, a few extra updates are seen, after months 75 updates have been counted
- The issue seems to affect other bindings as well: My Verisure binding (also communicating over an API) also gets duplicated sensor readings.
- The sensor burst readings happen at the expected time interval (every 5 min)
- Timeout expection sometimes happen (see second log below)

Example of logging:
![image](https://user-images.githubusercontent.com/9116407/192610810-f1381031-49ac-4ba1-959a-3bf28dde580e.png)

Logging with Trace enabled for the tellstick binding:
```
2022-09-12 19:17:30.305 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Updated sensors:[TellstickNetSensor [deviceId=1540384552, protocol=fineoffset, name=Basement, online=true, data=[DataTypeValue [dataType=temp, data=17.2, unit=°C], DataTypeValue [dataType=humidity, data=78, unit=%]], lastUpdated=1663002946, updated=false], TellstickNetSensor [deviceId=1540555924, protocol=fineoffset, name=Crawl space, online=true, data=[DataTypeValue [dataType=temp, data=16.4, unit=°C], DataTypeValue [dataType=humidity, data=82, unit=%]], lastUpdated=1663002427, updated=false], TellstickNetSensor [deviceId=1540556014, protocol=fineoffset, name=Outdoor, online=true, data=[DataTypeValue [dataType=temp, data=16.7, unit=°C], DataTypeValue [dataType=humidity, data=80, unit=%]], lastUpdated=1663002797, updated=false]]
2022-09-12 19:17:30.306 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensors, reset updated flag
2022-09-12 19:17:30.307 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensors, reset updated flag1
2022-09-12 19:17:30.308 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540384552, prev update 1663002946, new update 1663002946
2022-09-12 19:17:30.309 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540555924, prev update 1663002427, new update 1663002427
2022-09-12 19:17:30.309 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540556014, prev update 1663002797, new update 1663002797
2022-09-12 19:17:33.421 [TRACE] [nal.live.TelldusLiveDeviceController] - Request [http://api.telldus.com/xml/devices/list?supportedMethods=19] Response:<?xml version="1.0" encoding="utf-8" ?>
<devices>
	<device id="9945431" clientDeviceId="14" name="Carport Inner Lights" state="2" statevalue="" methods="3" metadataHash="" parametersHash="7c36bf492221e43b2065609d9b46fe7dd402eae8" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="10666457" clientDeviceId="15" name="Dusk Sensor" state="1" statevalue="" methods="3" metadataHash="" parametersHash="f5b96d8776b5945bc86e355d54bc1fab278fcdca" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="8597597" clientDeviceId="11" name="D&#246;rrklocka" state="2" statevalue="" methods="3" metadataHash="" parametersHash="18ad0b3f18aa1c4d9a24045e4c322f047b1d555f" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="7969581" clientDeviceId="6" name="Ek/V&#228;g-lampa" state="2" statevalue="" methods="3" metadataHash="" parametersHash="00c8346f30f4a05cba3c6bb783201d2d3b81cd0f" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="588090" clientDeviceId="2" name="Element h&#246;ger" state="2" statevalue="unde" methods="3" metadataHash="" parametersHash="673f0bf447780556fba79c0d476c11ac2eb12812" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="8468702" clientDeviceId="8" name="Element v&#228;nster" state="2" statevalue="" methods="3" metadataHash="" parametersHash="bafbd01080b4c074b19e0ff6974dc4e1eb49a460" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="9918531" clientDeviceId="13" name="Framd&#246;rr lampa" state="2" statevalue="" methods="3" metadataHash="" parametersHash="9d06f3925130649b157c489c14077eea7f33932e" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="9918526" clientDeviceId="12" name="K&#228;llarlampa" state="2" statevalue="" methods="3" metadataHash="" parametersHash="1828f54a12ce36b352f8202e3b7f12c507ec9b63" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="7969671" clientDeviceId="7" name="Lyktstolpe" state="2" statevalue="" methods="3" metadataHash="" parametersHash="bf7ea38193dd79b833ac27ac1f87216d4b5d4a8c" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
</devices>

2022-09-12 19:17:33.422 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device list [TellstickNetDevice [deviceId=9945431, name=Carport Inner Lights, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=10666457, name=Dusk Sensor, online=true, state=1, statevalue=, updated=false], TellstickNetDevice [deviceId=8597597, name=Dörrklocka, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=7969581, name=Ek/Väg-lampa, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=588090, name=Element höger, online=true, state=2, statevalue=unde, updated=false], TellstickNetDevice [deviceId=8468702, name=Element vänster, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=9918531, name=Framdörr lampa, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=9918526, name=Källarlampa, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=7969671, name=Lyktstolpe, online=true, state=2, statevalue=, updated=false]]
2022-09-12 19:17:33.423 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - updateDevices, Updating devices.
2022-09-12 19:17:33.424 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=9945431, name=Carport Inner Lights, online=true, state=2, statevalue=, updated=false] found at 0
2022-09-12 19:17:33.425 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=10666457, name=Dusk Sensor, online=true, state=1, statevalue=, updated=false] found at 1
2022-09-12 19:17:33.426 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=8597597, name=Dörrklocka, online=true, state=2, statevalue=, updated=false] found at 2
2022-09-12 19:17:33.427 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=7969581, name=Ek/Väg-lampa, online=true, state=2, statevalue=, updated=false] found at 3
2022-09-12 19:17:33.428 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=588090, name=Element höger, online=true, state=2, statevalue=unde, updated=false] found at 4
2022-09-12 19:17:33.428 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=8468702, name=Element vänster, online=true, state=2, statevalue=, updated=false] found at 5
2022-09-12 19:17:33.429 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=9918531, name=Framdörr lampa, online=true, state=2, statevalue=, updated=false] found at 6
2022-09-12 19:17:33.430 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=9918526, name=Källarlampa, online=true, state=2, statevalue=, updated=false] found at 7
2022-09-12 19:17:33.431 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=7969671, name=Lyktstolpe, online=true, state=2, statevalue=, updated=false] found at 8
2022-09-12 19:17:33.555 [TRACE] [nal.live.TelldusLiveDeviceController] - Request [http://api.telldus.com/xml/sensors/list?includeValues=1&includeScale=1&includeUnit=1] Response:<?xml version="1.0" encoding="utf-8" ?>
<sensors>
	<sensor id="1540384552" name="Basement" lastUpdated="1663002946" ignored="0" client="126139" clientName="Stanstorp" online="1" editable="1" battery="254" keepHistory="1" protocol="fineoffset" model="temperaturehumidity" sensorId="135" miscValues="null">
		<data name="temp" value="17.2" scale="0" lastUpdated="1663002946" max="24.4" maxTime="1606649223" min="6.9" minTime="1613117078" unit="&#176;C" />
		<data name="humidity" value="78" scale="0" lastUpdated="1663002946" max="95" maxTime="1626245828" min="40" minTime="1613204486" unit="%" />
	</sensor>
	<sensor id="1540555924" name="Crawl space" lastUpdated="1663003051" ignored="0" client="126139" clientName="Stanstorp" online="1" editable="1" battery="254" keepHistory="1" protocol="fineoffset" model="temperaturehumidity" sensorId="151" miscValues="null">
		<data name="temp" value="16.4" scale="0" lastUpdated="1663003051" max="23.7" maxTime="1607250269" min="6.9" minTime="1613117078" unit="&#176;C" />
		<data name="humidity" value="82" scale="0" lastUpdated="1663003051" max="89" maxTime="1657275991" min="45" minTime="1607251277" unit="%" />
	</sensor>
	<sensor id="1540556014" name="Outdoor" lastUpdated="1663002797" ignored="0" client="126139" clientName="Stanstorp" online="1" editable="1" battery="254" keepHistory="1" protocol="fineoffset" model="temperaturehumidity" sensorId="167" miscValues="{}">
		<data name="temp" value="16.7" scale="0" lastUpdated="1663002797" max="29.1" maxTime="1659616571" min="-10.3" minTime="1613117101" unit="&#176;C" />
		<data name="humidity" value="80" scale="0" lastUpdated="1663002797" max="99" maxTime="1607967323" min="34" minTime="1618835035" unit="%" />
	</sensor>
</sensors>

2022-09-12 19:17:33.556 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Updated sensors:[TellstickNetSensor [deviceId=1540384552, protocol=fineoffset, name=Basement, online=true, data=[DataTypeValue [dataType=temp, data=17.2, unit=°C], DataTypeValue [dataType=humidity, data=78, unit=%]], lastUpdated=1663002946, updated=false], TellstickNetSensor [deviceId=1540555924, protocol=fineoffset, name=Crawl space, online=true, data=[DataTypeValue [dataType=temp, data=16.4, unit=°C], DataTypeValue [dataType=humidity, data=82, unit=%]], lastUpdated=1663003051, updated=false], TellstickNetSensor [deviceId=1540556014, protocol=fineoffset, name=Outdoor, online=true, data=[DataTypeValue [dataType=temp, data=16.7, unit=°C], DataTypeValue [dataType=humidity, data=80, unit=%]], lastUpdated=1663002797, updated=false]]
2022-09-12 19:17:33.557 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensors, reset updated flag
2022-09-12 19:17:33.558 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensors, reset updated flag1
2022-09-12 19:17:33.558 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540384552, prev update 1663002946, new update 1663002946
2022-09-12 19:17:33.559 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540555924, prev update 1663002427, new update 1663003051
2022-09-12 19:17:33.559 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update for sensor:TellstickNetSensor [deviceId=1540555924, protocol=fineoffset, name=Crawl space, online=true, data=[DataTypeValue [dataType=temp, data=16.4, unit=°C], DataTypeValue [dataType=humidity, data=82, unit=%]], lastUpdated=1663003051, updated=false]
2022-09-12 19:17:33.560 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540556014, prev update 1663002797, new update 1663002797
2022-09-12 19:17:33.560 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540556014 or 1540556014
2022-09-12 19:17:33.561 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540556014 or 1540556014
2022-09-12 19:17:33.561 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540555924 or 1540555924
2022-09-12 19:17:33.563 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540555924 or 1540555924
2022-09-12 19:17:33.565 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540384552 or 1540384552
2022-09-12 19:17:33.566 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'SENSORCrawlspace_Temperature' updated to 16.4 °C
2022-09-12 19:17:33.566 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540384552 or 1540384552
2022-09-12 19:17:33.567 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:588090 or 588090
2022-09-12 19:17:33.568 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:588090 or 588090
2022-09-12 19:17:33.569 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'SENSORCrawlspace_Humidity' updated to 82 %
2022-09-12 19:17:33.568 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918531 or 9918531
2022-09-12 19:17:33.570 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918531 or 9918531
2022-09-12 19:17:33.571 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918526 or 9918526
2022-09-12 19:17:33.571 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918526 or 9918526
2022-09-12 19:17:33.572 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8468702 or 8468702
2022-09-12 19:17:33.572 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8468702 or 8468702
2022-09-12 19:17:33.573 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969581 or 7969581
2022-09-12 19:17:33.573 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969581 or 7969581
2022-09-12 19:17:33.574 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969671 or 7969671
2022-09-12 19:17:33.575 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969671 or 7969671
2022-09-12 19:17:33.575 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9945431 or 9945431
2022-09-12 19:17:33.576 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9945431 or 9945431
2022-09-12 19:17:33.576 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8597597 or 8597597
2022-09-12 19:17:33.576 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8597597 or 8597597
2022-09-12 19:17:33.577 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540555924 or 1540555924
2022-09-12 19:17:33.578 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540555924 or 1540555924
2022-09-12 19:17:33.579 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540556014 or 1540556014
2022-09-12 19:17:33.580 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'SENSORCrawlspace_Temperature' updated to 16.4 °C
2022-09-12 19:17:33.580 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540556014 or 1540556014
2022-09-12 19:17:33.581 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:588090 or 588090
2022-09-12 19:17:33.581 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'SENSORCrawlspace_Humidity' updated to 82 %
2022-09-12 19:17:33.581 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:588090 or 588090
2022-09-12 19:17:33.582 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918531 or 9918531
2022-09-12 19:17:33.582 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918531 or 9918531
2022-09-12 19:17:33.583 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8468702 or 8468702
2022-09-12 19:17:33.583 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8468702 or 8468702
2022-09-12 19:17:33.584 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540384552 or 1540384552
2022-09-12 19:17:33.584 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540384552 or 1540384552
2022-09-12 19:17:33.584 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969581 or 7969581
2022-09-12 19:17:33.585 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969581 or 7969581
2022-09-12 19:17:33.585 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969671 or 7969671
2022-09-12 19:17:33.585 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:7969671 or 7969671
2022-09-12 19:17:33.586 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918526 or 9918526
2022-09-12 19:17:33.586 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9918526 or 9918526
2022-09-12 19:17:33.586 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9945431 or 9945431
2022-09-12 19:17:33.587 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:9945431 or 9945431
2022-09-12 19:17:33.587 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8597597 or 8597597
2022-09-12 19:17:33.587 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:8597597 or 8597597
2022-09-12 19:17:33.588 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540555924 or 1540555924
2022-09-12 19:17:33.589 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540555924 or 1540555924
2022-09-12 19:17:33.590 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540556014 or 1540556014
2022-09-12 19:17:33.591 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'SENSORCrawlspace_Temperature' updated to 16.4 °C
2022-09-12 19:17:33.591 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540556014 or 1540556014
2022-09-12 19:17:33.592 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:588090 or 588090
2022-09-12 19:17:33.592 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:switch:5cb57369ab:588090 or 588090
2022-09-12 19:17:33.593 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating states of (SENSOR Crawl space (1540555924) id: tellstick:sensor:5cb57369ab:1540384552 or 1540384552
2022-09-12 19:17:33.593 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'SENSORCrawlspace_Humidity' updated to 82 %

```
TimeoutException error message that sometimes appear:
```
2022-09-11 20:20:02.903 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=7969671, name=Lyktstolpe, online=true, state=2, statevalue=, updated=false] found at 8
2022-09-11 20:20:07.906 [WARN ] [nal.live.TelldusLiveDeviceController] - TimeoutException error in get
java.util.concurrent.TimeoutException: null
	at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1886) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2021) ~[?:?]
	at org.asynchttpclient.netty.NettyResponseFuture.get(NettyResponseFuture.java:206) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:309) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateSensors(TelldusLiveBridgeHandler.java:181) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:117) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
2022-09-11 20:20:08.229 [WARN ] [nal.live.TelldusLiveDeviceController] - Encoding error in get
javax.xml.bind.UnmarshalException: null
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.handleStreamException(UnmarshallerImpl.java:453) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:387) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateSensors(TelldusLiveBridgeHandler.java:181) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:117) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: com.sun.istack.SAXParseException2: unexpected element (uri:"", local:"html"). Expected elements are <{}data>,<{}sensor>,<{}sensors>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	... 11 more
Caused by: javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}data>,<{}sensor>,<{}sensors>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	... 11 more
2022-09-11 20:20:08.236 [WARN ] [nal.live.TelldusLiveDeviceController] - Request [http://api.telldus.com/xml/sensors/list?includeValues=1&includeScale=1&includeUnit=1] Failure:null
2022-09-11 20:20:08.238 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:telldus-live:5cb57369ab' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.238 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540555924 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.240 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540556014 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.240 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9918531 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.241 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 588090 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.242 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540384552 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.242 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540555924' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.242 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9918526 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.242 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 7969581 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.243 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 8468702 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.243 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9945431 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.244 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 8597597 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.245 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9918531' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.246 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540556014' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.247 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9918526' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.243 [ERROR] [ternal.live.TelldusLiveBridgeHandler] - Failed to update
org.openhab.binding.tellstick.internal.live.TelldusLiveException: null
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:294) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateSensors(TelldusLiveBridgeHandler.java:181) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:117) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: javax.xml.bind.UnmarshalException
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.handleStreamException(UnmarshallerImpl.java:453) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:387) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	... 8 more
Caused by: com.sun.istack.SAXParseException2: unexpected element (uri:"", local:"html"). Expected elements are <{}data>,<{}sensor>,<{}sensors>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	... 8 more
Caused by: javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}data>,<{}sensor>,<{}sensors>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	... 8 more
2022-09-11 20:20:08.248 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:7969581' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-09-11 20:20:08.244 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 7969671 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
```

<!-- If describing a bug, tell us what happens instead of the expected behavior -->
<!-- Include related log information (preferably debug level) and related configs -->
<!-- Use a file attachment for log and config information longer than a few lines -->
<!-- Enclose multi-line log/code snippets with ``` on new lines for proper formatting -->
<!-- If suggesting a change/improvement, explain the difference from current behavior -->
<!-- For improvements, discuss at community.openhab.org first and include link to topic -->

## Possible Solution
Input from rossko57 in the community thread:
"This feels more like faulty task re-scheduling after query/response, in some circumstance or other (quite possibly a remote timeout, though that would be the trigger not the root cause).
That would be a binding issue."

## Steps to Reproduce (for Bugs)
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->
1. Lower tellstick refresh interval to 3s
2. Wait for a few hours
3. Observe in the log that sensor values are duplicated

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->
The excess points in the influxdb database makes graphs very slow.

A potential work around would be to only store changes in the database, but then I loose information about when sensor values are missing. E.g. in the crawl space, the humidity might not change for a whole day, which would give an empty graph.

## Your Environment
Hardware: Raspberry pi 4, 4GB
OS: openhabian
openHAB version: 3.3.0.M3
Persistence: Influxdb
Configuration method: GUI only, no files. Semantic model is used
Tellstick usage: Live (cloud API)

Link to community thread:
https://community.openhab.org/t/tellstick-binding-item-updates-get-duplicated-around-75-times-after-lowering-refresh-interval/138634

New finding:
There is another error, which I must have missed before, called "Encoding error" (see below).

The occurances differ from time to time, but in one day I saw almost 600 errors, and afterwards there were 222 duplications!

More interestingly, starting from a freshly restarted openhab without any duplications, each error caused exactly one added duplication! (at least the two first occurances, then it get's messy with many errors on top of each other).

So hopefully if we solve this error, we can get rid of the duplications.

Any idea what is causing it?

```
2022-10-21 10:01:41.884 [WARN ] [nal.live.TelldusLiveDeviceController] - Encoding error in get
javax.xml.bind.UnmarshalException: null
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.handleStreamException(UnmarshallerImpl.java:453) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:387) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateDevices(TelldusLiveBridgeHandler.java:129) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:116) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: com.sun.istack.SAXParseException2: unexpected element (uri:"", local:"html"). Expected elements are <{}devices>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	... 11 more
Caused by: javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}devices>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	... 11 more
2022-10-21 10:01:41.890 [WARN ] [nal.live.TelldusLiveDeviceController] - Request [http://api.telldus.com/xml/devices/list?supportedMethods=19] Failure:null
2022-10-21 10:01:41.893 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:telldus-live:5cb57369ab' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.894 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 7969581 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.893 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 7969671 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.895 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 588090 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.895 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540555924 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.896 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9945431 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.896 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540384552 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.896 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9918526 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.897 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:7969581' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.897 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540556014 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.897 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 8597597 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.898 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 8468702 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.898 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9918531 bridgeStatusChanged: OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.898 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:7969671' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.900 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:588090' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.901 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540555924' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.902 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9918526' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.895 [ERROR] [ternal.live.TelldusLiveBridgeHandler] - Failed to update
org.openhab.binding.tellstick.internal.live.TelldusLiveException: null
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:294) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.updateDevices(TelldusLiveBridgeHandler.java:129) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler.refreshDeviceList(TelldusLiveBridgeHandler.java:116) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: javax.xml.bind.UnmarshalException
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.handleStreamException(UnmarshallerImpl.java:453) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:387) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	... 8 more
Caused by: com.sun.istack.SAXParseException2: unexpected element (uri:"", local:"html"). Expected elements are <{}devices>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	... 8 more
Caused by: javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"html"). Expected elements are <{}devices>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportUnexpectedChildElement(Loader.java:94) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext$DefaultRootLoader.childElement(UnmarshallingContext.java:1117) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext._startElement(UnmarshallingContext.java:542) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.startElement(UnmarshallingContext.java:524) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.handleStartElement(StAXStreamConnector.java:216) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.StAXStreamConnector.bridge(StAXStreamConnector.java:150) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:385) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:356) ~[jaxb-runtime-2.3.3.jar:2.3.3]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.innerCallRest(TelldusLiveDeviceController.java:319) ~[?:?]
	at org.openhab.binding.tellstick.internal.live.TelldusLiveDeviceController.callRestMethod(TelldusLiveDeviceController.java:283) ~[?:?]
	... 8 more
2022-10-21 10:01:41.902 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540384552' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.903 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540556014' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.904 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9945431' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.905 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:8597597' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.905 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:8468702' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:01:41.906 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9918531' changed from ONLINE to OFFLINE (COMMUNICATION_ERROR)
2022-10-21 10:02:00.023 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'HeatingControl_HeatingPower' updated to 0 W
2022-10-21 10:02:00.188 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'ElectricityMeter_TotalCurrentPower' updated to 0.099
2022-10-21 10:02:00.190 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'ElectricityMeter_TotalCurrentPower' changed from 0.1 kW to 0.099 kW
2022-10-21 10:02:10.373 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'ElectricityMeter_TotalCurrentPower' updated to 0.1
2022-10-21 10:02:10.375 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'ElectricityMeter_TotalCurrentPower' changed from 0.099 kW to 0.1 kW
2022-10-21 10:02:20.523 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'ElectricityMeter_TotalAccumulativePower' updated to 4325.415
2022-10-21 10:02:20.530 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'ElectricityMeter_TotalCurrentPower' updated to 0.099
2022-10-21 10:02:20.531 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'ElectricityMeter_TotalAccumulativePower' changed from 4325.414 kWh to 4325.415 kWh
2022-10-21 10:02:20.534 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'ElectricityMeter_TotalCurrentPower' changed from 0.1 kW to 0.099 kW
2022-10-21 10:02:30.669 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'ElectricityMeter_TotalCurrentPower' updated to 0.1
2022-10-21 10:02:30.670 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'ElectricityMeter_TotalCurrentPower' changed from 0.099 kW to 0.1 kW
2022-10-21 10:02:43.080 [TRACE] [nal.live.TelldusLiveDeviceController] - Request [http://api.telldus.com/xml/devices/list?supportedMethods=19] Response:<?xml version="1.0" encoding="utf-8" ?>
<devices>
	<device id="9945431" clientDeviceId="14" name="Carport Inner Lights" state="2" statevalue="" methods="3" metadataHash="" parametersHash="7c36bf492221e43b2065609d9b46fe7dd402eae8" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="10666457" clientDeviceId="15" name="Dusk Sensor" state="2" statevalue="" methods="3" metadataHash="" parametersHash="f5b96d8776b5945bc86e355d54bc1fab278fcdca" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="8597597" clientDeviceId="11" name="D&#246;rrklocka" state="2" statevalue="" methods="3" metadataHash="" parametersHash="18ad0b3f18aa1c4d9a24045e4c322f047b1d555f" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="7969581" clientDeviceId="6" name="Ek/V&#228;g-lampa" state="2" statevalue="" methods="3" metadataHash="" parametersHash="00c8346f30f4a05cba3c6bb783201d2d3b81cd0f" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="588090" clientDeviceId="2" name="Element h&#246;ger" state="2" statevalue="unde" methods="3" metadataHash="" parametersHash="673f0bf447780556fba79c0d476c11ac2eb12812" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="8468702" clientDeviceId="8" name="Element v&#228;nster" state="2" statevalue="" methods="3" metadataHash="" parametersHash="bafbd01080b4c074b19e0ff6974dc4e1eb49a460" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="9918531" clientDeviceId="13" name="Framd&#246;rr lampa" state="2" statevalue="" methods="3" metadataHash="" parametersHash="9d06f3925130649b157c489c14077eea7f33932e" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="9918526" clientDeviceId="12" name="K&#228;llarlampa" state="2" statevalue="" methods="3" metadataHash="" parametersHash="1828f54a12ce36b352f8202e3b7f12c507ec9b63" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
	<device id="7969671" clientDeviceId="7" name="Lyktstolpe" state="2" statevalue="" methods="3" metadataHash="" parametersHash="bf7ea38193dd79b833ac27ac1f87216d4b5d4a8c" type="device" client="126139" clientName="Stanstorp" online="1" editable="1" ignored="0" />
</devices>

2022-10-21 10:02:43.082 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device list [TellstickNetDevice [deviceId=9945431, name=Carport Inner Lights, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=10666457, name=Dusk Sensor, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=8597597, name=Dörrklocka, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=7969581, name=Ek/Väg-lampa, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=588090, name=Element höger, online=true, state=2, statevalue=unde, updated=false], TellstickNetDevice [deviceId=8468702, name=Element vänster, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=9918531, name=Framdörr lampa, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=9918526, name=Källarlampa, online=true, state=2, statevalue=, updated=false], TellstickNetDevice [deviceId=7969671, name=Lyktstolpe, online=true, state=2, statevalue=, updated=false]]
2022-10-21 10:02:43.083 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - updateDevices, Updating devices.
2022-10-21 10:02:43.085 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=9945431, name=Carport Inner Lights, online=true, state=2, statevalue=, updated=false] found at 0
2022-10-21 10:02:43.085 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=10666457, name=Dusk Sensor, online=true, state=2, statevalue=, updated=false] found at 1
2022-10-21 10:02:43.086 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=8597597, name=Dörrklocka, online=true, state=2, statevalue=, updated=false] found at 2
2022-10-21 10:02:43.087 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=7969581, name=Ek/Väg-lampa, online=true, state=2, statevalue=, updated=false] found at 3
2022-10-21 10:02:43.087 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=588090, name=Element höger, online=true, state=2, statevalue=unde, updated=false] found at 4
2022-10-21 10:02:43.088 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=8468702, name=Element vänster, online=true, state=2, statevalue=, updated=false] found at 5
2022-10-21 10:02:43.088 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=9918531, name=Framdörr lampa, online=true, state=2, statevalue=, updated=false] found at 6
2022-10-21 10:02:43.089 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=9918526, name=Källarlampa, online=true, state=2, statevalue=, updated=false] found at 7
2022-10-21 10:02:43.089 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Device:TellstickNetDevice [deviceId=7969671, name=Lyktstolpe, online=true, state=2, statevalue=, updated=false] found at 8
2022-10-21 10:02:44.418 [TRACE] [nal.live.TelldusLiveDeviceController] - Request [http://api.telldus.com/xml/sensors/list?includeValues=1&includeScale=1&includeUnit=1] Response:<?xml version="1.0" encoding="utf-8" ?>
<sensors>
	<sensor id="1540384552" name="Basement" lastUpdated="1666339230" ignored="0" client="126139" clientName="Stanstorp" online="1" editable="1" battery="254" keepHistory="1" protocol="fineoffset" model="temperaturehumidity" sensorId="135" miscValues="null">
		<data name="temp" value="12.7" scale="0" lastUpdated="1666339230" max="24.4" maxTime="1606649223" min="6.9" minTime="1613117078" unit="&#176;C" />
		<data name="humidity" value="77" scale="0" lastUpdated="1666339230" max="95" maxTime="1626245828" min="40" minTime="1613204486" unit="%" />
	</sensor>
	<sensor id="1540555924" name="Crawl space" lastUpdated="1666338973" ignored="0" client="126139" clientName="Stanstorp" online="1" editable="1" battery="254" keepHistory="1" protocol="fineoffset" model="temperaturehumidity" sensorId="151" miscValues="null">
		<data name="temp" value="13.2" scale="0" lastUpdated="1666338973" max="23.7" maxTime="1607250269" min="6.9" minTime="1613117078" unit="&#176;C" />
		<data name="humidity" value="78" scale="0" lastUpdated="1666338973" max="89" maxTime="1657275991" min="45" minTime="1607251277" unit="%" />
	</sensor>
	<sensor id="1540556014" name="Outdoor" lastUpdated="1666338789" ignored="0" client="126139" clientName="Stanstorp" online="1" editable="1" battery="254" keepHistory="1" protocol="fineoffset" model="temperaturehumidity" sensorId="167" miscValues="{}">
		<data name="temp" value="9.2" scale="0" lastUpdated="1666338789" max="29.1" maxTime="1659616571" min="-10.3" minTime="1613117101" unit="&#176;C" />
		<data name="humidity" value="97" scale="0" lastUpdated="1666338789" max="99" maxTime="1607967323" min="34" minTime="1618835035" unit="%" />
	</sensor>
</sensors>

2022-10-21 10:02:44.420 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Updated sensors:[TellstickNetSensor [deviceId=1540384552, protocol=fineoffset, name=Basement, online=true, data=[DataTypeValue [dataType=temp, data=12.7, unit=°C], DataTypeValue [dataType=humidity, data=77, unit=%]], lastUpdated=1666339230, updated=false], TellstickNetSensor [deviceId=1540555924, protocol=fineoffset, name=Crawl space, online=true, data=[DataTypeValue [dataType=temp, data=13.2, unit=°C], DataTypeValue [dataType=humidity, data=78, unit=%]], lastUpdated=1666338973, updated=false], TellstickNetSensor [deviceId=1540556014, protocol=fineoffset, name=Outdoor, online=true, data=[DataTypeValue [dataType=temp, data=9.2, unit=°C], DataTypeValue [dataType=humidity, data=97, unit=%]], lastUpdated=1666338789, updated=false]]
2022-10-21 10:02:44.420 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensors, reset updated flag
2022-10-21 10:02:44.421 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensors, reset updated flag1
2022-10-21 10:02:44.422 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540384552, prev update 1666339230, new update 1666339230
2022-10-21 10:02:44.423 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540555924, prev update 1666338973, new update 1666338973
2022-10-21 10:02:44.423 [DEBUG] [ternal.live.TelldusLiveBridgeHandler] - Update sensor 1540556014, prev update 1666338789, new update 1666338789
2022-10-21 10:02:44.424 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 7969581 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.424 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 588090 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.424 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9945431 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 588090, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.424 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 7969671 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 7969581, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 9945431, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.424 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540555924 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Element höger (588090) id: tellstick:switch:5cb57369ab:588090
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Carport Inner Lights (9945431) id: tellstick:switch:5cb57369ab:9945431
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 1540555924, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 7969671, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.426 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Lyktstolpe (7969671) id: tellstick:switch:5cb57369ab:7969671
2022-10-21 10:02:44.429 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540384552 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.429 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9918526 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.429 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 1540384552, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.430 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 1540556014 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.430 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 1540556014, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.425 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Ek/Väg-lampa (7969581) id: tellstick:switch:5cb57369ab:7969581
2022-10-21 10:02:44.433 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 8597597 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.433 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:telldus-live:5cb57369ab' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.430 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 9918526, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.434 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Källarlampa (9918526) id: tellstick:switch:5cb57369ab:9918526
2022-10-21 10:02:44.435 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 9918531 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.436 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 9918531, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.436 [DEBUG] [ternal.handler.TelldusDevicesHandler] - device: 8468702 bridgeStatusChanged: ONLINE
2022-10-21 10:02:44.437 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Framdörr lampa (9918531) id: tellstick:switch:5cb57369ab:9918531
2022-10-21 10:02:44.437 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 8468702, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.433 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Init bridge for 8597597, bridge:org.openhab.binding.tellstick.internal.live.TelldusLiveBridgeHandler@154f7ec
2022-10-21 10:02:44.438 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Dörrklocka (8597597) id: tellstick:switch:5cb57369ab:8597597
2022-10-21 10:02:44.438 [DEBUG] [ternal.handler.TelldusDevicesHandler] - Updating state of DEVICE Element vänster (8468702) id: tellstick:switch:5cb57369ab:8468702
2022-10-21 10:02:44.439 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'VerandaRadiatorRight_Switch' updated to OFF
2022-10-21 10:02:44.442 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'CarportInnerLight_Switch' updated to OFF
2022-10-21 10:02:44.443 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:588090' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.444 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9945431' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.445 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'LightPole_Switch' updated to OFF
2022-10-21 10:02:44.446 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:7969671' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.446 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:9945431' has been updated.
2022-10-21 10:02:44.447 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:588090' has been updated.
2022-10-21 10:02:44.448 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:7969671' has been updated.
2022-10-21 10:02:44.448 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:sensor:5cb57369ab:1540384552' has been updated.
2022-10-21 10:02:44.449 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540384552' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.449 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:sensor:5cb57369ab:1540555924' has been updated.
2022-10-21 10:02:44.451 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'OakRoadLight_Switch' updated to OFF
2022-10-21 10:02:44.451 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:7969581' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.452 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:7969581' has been updated.
2022-10-21 10:02:44.453 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540555924' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.453 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:sensor:5cb57369ab:1540556014' has been updated.
2022-10-21 10:02:44.454 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'BasementLight_Switch' updated to OFF
2022-10-21 10:02:44.455 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:sensor:5cb57369ab:1540556014' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.456 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9918526' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.457 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'FrontdoorLight_Switch' updated to OFF
2022-10-21 10:02:44.458 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:9918526' has been updated.
2022-10-21 10:02:44.459 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'Doorbell_Switch' updated to OFF
2022-10-21 10:02:44.460 [INFO ] [openhab.event.ItemStateEvent        ] - Item 'VerandaRadiatorLeft_Switch' updated to OFF
2022-10-21 10:02:44.461 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:8597597' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.461 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:8468702' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.462 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:8597597' has been updated.
2022-10-21 10:02:44.462 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:8468702' has been updated.
2022-10-21 10:02:44.463 [INFO ] [ab.event.ThingStatusInfoChangedEvent] - Thing 'tellstick:switch:5cb57369ab:9918531' changed from OFFLINE (COMMUNICATION_ERROR) to ONLINE
2022-10-21 10:02:44.464 [INFO ] [openhab.event.ThingUpdatedEvent     ] - Thing 'tellstick:switch:5cb57369ab:9918531' has been updated.
```

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/TellstickHandlerFactory.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/TellstickHandlerFactory.java
@@ -31,6 +31,7 @@ import org.openhab.core.thing.ThingTypeUID;
 import org.openhab.core.thing.binding.BaseThingHandlerFactory;
 import org.openhab.core.thing.binding.ThingHandler;
 import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.framework.ServiceRegistration;
 import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
@@ -47,8 +48,9 @@ import org.slf4j.LoggerFactory;
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.tellstick")
 public class TellstickHandlerFactory extends BaseThingHandlerFactory {
     private final Logger logger = LoggerFactory.getLogger(TellstickHandlerFactory.class);
-    private TellstickDiscoveryService discoveryService = null;
     private final HttpClient httpClient;
+    private TellstickDiscoveryService discoveryService = null;
+    private ServiceRegistration<DiscoveryService> discoveryServiceRegistration = null;
 
     @Activate
     public TellstickHandlerFactory(@Reference HttpClientFactory httpClientFactory) {
@@ -60,13 +62,33 @@ public class TellstickHandlerFactory extends BaseThingHandlerFactory {
         return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
     }
 
-    private void registerDeviceDiscoveryService(TelldusBridgeHandler tellstickBridgeHandler) {
-        if (discoveryService == null) {
-            discoveryService = new TellstickDiscoveryService(tellstickBridgeHandler);
-            discoveryService.activate();
-            bundleContext.registerService(DiscoveryService.class.getName(), discoveryService, new Hashtable<>());
+    private synchronized void registerDeviceDiscoveryService(TelldusBridgeHandler tellstickBridgeHandler) {
+        TellstickDiscoveryService service = discoveryService;
+        if (service == null) {
+            service = new TellstickDiscoveryService(tellstickBridgeHandler);
+            service.activate();
+            discoveryService = service;
+            discoveryServiceRegistration = (ServiceRegistration<DiscoveryService>) bundleContext
+                    .registerService(DiscoveryService.class.getName(), service, new Hashtable<>());
         } else {
-            discoveryService.addBridgeHandler(tellstickBridgeHandler);
+            service.addBridgeHandler(tellstickBridgeHandler);
+        }
+    }
+
+    private synchronized void unregisterDeviceDiscoveryService(TelldusBridgeHandler tellstickBridgeHandler) {
+        TellstickDiscoveryService service = discoveryService;
+        if (service != null) {
+            if (service.isOnlyForOneBridgeHandler()) {
+                service.deactivate();
+                discoveryService = null;
+                ServiceRegistration<DiscoveryService> serviceReg = discoveryServiceRegistration;
+                if (serviceReg != null) {
+                    serviceReg.unregister();
+                    discoveryServiceRegistration = null;
+                }
+            } else {
+                service.removeBridgeHandler(tellstickBridgeHandler);
+            }
         }
     }
 
@@ -89,6 +111,13 @@ public class TellstickHandlerFactory extends BaseThingHandlerFactory {
         } else {
             logger.debug("ThingHandler not found for {}", thing.getThingTypeUID());
             return null;
+        }
+    }
+
+    @Override
+    protected void removeHandler(ThingHandler thingHandler) {
+        if (thingHandler instanceof TelldusBridgeHandler) {
+            unregisterDeviceDiscoveryService((TelldusBridgeHandler) thingHandler);
         }
     }
 }

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/TellstickHandlerFactory.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/TellstickHandlerFactory.java
@@ -16,6 +16,8 @@ import static org.openhab.binding.tellstick.internal.TellstickBindingConstants.*
 
 import java.util.Hashtable;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.jetty.client.HttpClient;
 import org.openhab.binding.tellstick.internal.core.TelldusCoreBridgeHandler;
 import org.openhab.binding.tellstick.internal.discovery.TellstickDiscoveryService;
@@ -44,13 +46,15 @@ import org.slf4j.LoggerFactory;
  *
  * @author Jarle Hjortland - Initial contribution
  * @author Jan Gustafsson - Adding support for local API
+ * @author Laurent Garnier - fix discovery service registering/unregistering
  */
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.tellstick")
+@NonNullByDefault
 public class TellstickHandlerFactory extends BaseThingHandlerFactory {
     private final Logger logger = LoggerFactory.getLogger(TellstickHandlerFactory.class);
     private final HttpClient httpClient;
-    private TellstickDiscoveryService discoveryService = null;
-    private ServiceRegistration<DiscoveryService> discoveryServiceRegistration = null;
+    private @Nullable TellstickDiscoveryService discoveryService = null;
+    private @Nullable ServiceRegistration<DiscoveryService> discoveryServiceRegistration = null;
 
     @Activate
     public TellstickHandlerFactory(@Reference HttpClientFactory httpClientFactory) {
@@ -93,7 +97,7 @@ public class TellstickHandlerFactory extends BaseThingHandlerFactory {
     }
 
     @Override
-    protected ThingHandler createHandler(Thing thing) {
+    protected @Nullable ThingHandler createHandler(Thing thing) {
         if (thing.getThingTypeUID().equals(TELLDUSCOREBRIDGE_THING_TYPE)) {
             TelldusCoreBridgeHandler handler = new TelldusCoreBridgeHandler((Bridge) thing);
             registerDeviceDiscoveryService(handler);

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java
@@ -160,7 +160,6 @@ public class TelldusCoreBridgeHandler extends BaseBridgeHandler
 
                 }
             }
-
         } catch (SupportedMethodsException e) {
             logger.error("Failed to get devices ", e);
         }
@@ -186,7 +185,8 @@ public class TelldusCoreBridgeHandler extends BaseBridgeHandler
         if (deviceStatusListener == null) {
             throw new IllegalArgumentException("It's not allowed to pass a null deviceStatusListener.");
         }
-        return deviceStatusListeners.add(deviceStatusListener);
+        return deviceStatusListeners.contains(deviceStatusListener) ? false
+                : deviceStatusListeners.add(deviceStatusListener);
     }
 
     @Override

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java
@@ -16,8 +16,9 @@ import java.util.Calendar;
 import java.util.Collections;
 import java.util.Date;
 import java.util.List;
+import java.util.Set;
 import java.util.Vector;
-import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ConcurrentHashMap;
 
 import org.openhab.binding.tellstick.internal.conf.TellstickBridgeConfiguration;
 import org.openhab.binding.tellstick.internal.handler.DeviceStatusListener;
@@ -66,7 +67,7 @@ public class TelldusCoreBridgeHandler extends BaseBridgeHandler
     private List<TellstickSensor> sensorList = new Vector<>();
     private TellstickEventHandler eventHandler;
     private static boolean initialized = false;
-    private List<DeviceStatusListener> deviceStatusListeners = new CopyOnWriteArrayList<>();
+    private Set<DeviceStatusListener> deviceStatusListeners = ConcurrentHashMap.newKeySet();
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
@@ -185,8 +186,7 @@ public class TelldusCoreBridgeHandler extends BaseBridgeHandler
         if (deviceStatusListener == null) {
             throw new IllegalArgumentException("It's not allowed to pass a null deviceStatusListener.");
         }
-        return deviceStatusListeners.contains(deviceStatusListener) ? false
-                : deviceStatusListeners.add(deviceStatusListener);
+        return deviceStatusListeners.add(deviceStatusListener);
     }
 
     @Override

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/discovery/TellstickDiscoveryService.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/discovery/TellstickDiscoveryService.java
@@ -205,4 +205,13 @@ public class TellstickDiscoveryService extends AbstractDiscoveryService implemen
         telldusBridgeHandlers.add(tellstickBridgeHandler);
         tellstickBridgeHandler.registerDeviceStatusListener(this);
     }
+
+    public void removeBridgeHandler(TelldusBridgeHandler tellstickBridgeHandler) {
+        telldusBridgeHandlers.remove(tellstickBridgeHandler);
+        tellstickBridgeHandler.unregisterDeviceStatusListener(this);
+    }
+
+    public boolean isOnlyForOneBridgeHandler() {
+        return telldusBridgeHandlers.size() == 1;
+    }
 }

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/handler/TelldusDevicesHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/handler/TelldusDevicesHandler.java
@@ -166,7 +166,7 @@ public class TelldusDevicesHandler extends BaseThingHandler implements DeviceSta
             if (bridge != null) {
                 bridgeStatusChanged(bridge.getStatusInfo());
             } else {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge defined");
             }
         } else {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/handler/TelldusDevicesHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/handler/TelldusDevicesHandler.java
@@ -151,24 +151,26 @@ public class TelldusDevicesHandler extends BaseThingHandler implements DeviceSta
     public void initialize() {
         Configuration config = getConfig();
         logger.debug("Initialize TelldusDeviceHandler {}. class {}", config, config.getClass());
-        final Object configDeviceId = config.get(TellstickBindingConstants.DEVICE_ID);
-        if (configDeviceId != null) {
-            deviceId = configDeviceId.toString();
-        } else {
-            logger.debug("Initialized TellStick device missing serialNumber configuration... troubles ahead");
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
-        }
         final Boolean isADimmer = (Boolean) config.get(TellstickBindingConstants.DEVICE_ISDIMMER);
         if (isADimmer != null) {
-            this.isDimmer = isADimmer;
+            isDimmer = isADimmer;
         }
         final BigDecimal repeatCount = (BigDecimal) config.get(TellstickBindingConstants.DEVICE_RESEND_COUNT);
         if (repeatCount != null) {
             resend = repeatCount.intValue();
         }
-        Bridge bridge = getBridge();
-        if (bridge != null) {
-            bridgeStatusChanged(bridge.getStatusInfo());
+        final Object configDeviceId = config.get(TellstickBindingConstants.DEVICE_ID);
+        if (configDeviceId != null) {
+            deviceId = configDeviceId.toString();
+            Bridge bridge = getBridge();
+            if (bridge != null) {
+                bridgeStatusChanged(bridge.getStatusInfo());
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Missing serialNumber configuration");
         }
     }
 
@@ -180,7 +182,7 @@ public class TelldusDevicesHandler extends BaseThingHandler implements DeviceSta
                 Bridge localBridge = getBridge();
                 if (localBridge != null) {
                     TelldusBridgeHandler telldusBridgeHandler = (TelldusBridgeHandler) localBridge.getHandler();
-                    logger.debug("Init bridge for {}, bridge:{}", deviceId, telldusBridgeHandler);
+                    logger.debug("Init device {}, bridge:{}", deviceId, telldusBridgeHandler);
                     if (telldusBridgeHandler != null) {
                         this.bridgeHandler = telldusBridgeHandler;
                         this.bridgeHandler.registerDeviceStatusListener(this);
@@ -208,12 +210,21 @@ public class TelldusDevicesHandler extends BaseThingHandler implements DeviceSta
                     }
                 }
             } catch (Exception e) {
-                logger.error("Failed to init bridge for {}", deviceId, e);
+                logger.warn("Failed to init device {}", deviceId, e);
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR);
             }
         } else {
-            updateStatus(ThingStatus.OFFLINE, bridgeStatusInfo.getStatusDetail());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
         }
+    }
+
+    @Override
+    public void dispose() {
+        TelldusBridgeHandler bridgeHandler = getTellstickBridgeHandler();
+        if (bridgeHandler != null) {
+            bridgeHandler.unregisterDeviceStatusListener(this);
+        }
+        super.dispose();
     }
 
     private Device getDevice(TelldusBridgeHandler tellHandler, String deviceId) {

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveBridgeHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveBridgeHandler.java
@@ -12,10 +12,11 @@
  */
 package org.openhab.binding.tellstick.internal.live;
 
-import java.time.LocalDateTime;
-import java.time.temporal.ChronoUnit;
+import java.time.Duration;
+import java.time.Instant;
 import java.util.List;
-import java.util.Vector;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
@@ -60,11 +61,11 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
     private TellstickNetDevices deviceList = null;
     private TellstickNetSensors sensorList = null;
     private TelldusLiveDeviceController controller = new TelldusLiveDeviceController();
-    private List<DeviceStatusListener> deviceStatusListeners = new Vector<>();
+    private Set<DeviceStatusListener> deviceStatusListeners = ConcurrentHashMap.newKeySet();
 
     private int nbRefresh;
     private long sumRefreshDuration;
-    private long minRefreshDuration = 1000000;
+    private long minRefreshDuration = 1_000_000;
     private long maxRefreshDuration;
 
     public TelldusLiveBridgeHandler(Bridge bridge) {
@@ -118,7 +119,7 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
     }
 
     synchronized void refreshDeviceList() {
-        LocalDateTime start = LocalDateTime.now();
+        Instant start = Instant.now();
         try {
             updateDevices(deviceList);
             updateSensors(sensorList);
@@ -127,14 +128,14 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
             logger.warn("Failed to update", e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
         }
-        monitorAdditionalRefresh(start, LocalDateTime.now());
+        monitorAdditionalRefresh(start, Instant.now());
     }
 
-    private void monitorAdditionalRefresh(LocalDateTime start, LocalDateTime end) {
+    private void monitorAdditionalRefresh(Instant start, Instant end) {
         if (!logger.isDebugEnabled()) {
             return;
         }
-        long duration = ChronoUnit.MILLIS.between(start, end);
+        long duration = Duration.between(start, end).toMillis();
         sumRefreshDuration += duration;
         nbRefresh++;
         if (duration < minRefreshDuration) {
@@ -283,8 +284,7 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
         if (deviceStatusListener == null) {
             throw new IllegalArgumentException("It's not allowed to pass a null deviceStatusListener.");
         }
-        return deviceStatusListeners.contains(deviceStatusListener) ? false
-                : deviceStatusListeners.add(deviceStatusListener);
+        return deviceStatusListeners.add(deviceStatusListener);
     }
 
     @Override

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveBridgeHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveBridgeHandler.java
@@ -12,6 +12,8 @@
  */
 package org.openhab.binding.tellstick.internal.live;
 
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
 import java.util.List;
 import java.util.Vector;
 import java.util.concurrent.ScheduledFuture;
@@ -51,6 +53,8 @@ import org.tellstick.device.iface.Device;
  */
 public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements TelldusBridgeHandler {
 
+    private static final int REFRESH_DELAY = 10;
+
     private final Logger logger = LoggerFactory.getLogger(TelldusLiveBridgeHandler.class);
 
     private TellstickNetDevices deviceList = null;
@@ -58,7 +62,10 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
     private TelldusLiveDeviceController controller = new TelldusLiveDeviceController();
     private List<DeviceStatusListener> deviceStatusListeners = new Vector<>();
 
-    private static final int REFRESH_DELAY = 10;
+    private int nbRefresh;
+    private long sumRefreshDuration;
+    private long minRefreshDuration = 1000000;
+    private long maxRefreshDuration;
 
     public TelldusLiveBridgeHandler(Bridge bridge) {
         super(bridge);
@@ -88,9 +95,8 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
         this.controller = new TelldusLiveDeviceController();
         this.controller.connectHttpClient(configuration.publicKey, configuration.privateKey, configuration.token,
                 configuration.tokenSecret);
+        updateStatus(ThingStatus.UNKNOWN);
         startAutomaticRefresh(configuration.refreshInterval);
-        refreshDeviceList();
-        updateStatus(ThingStatus.ONLINE);
     }
 
     private synchronized void startAutomaticRefresh(long refreshInterval) {
@@ -112,17 +118,36 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
     }
 
     synchronized void refreshDeviceList() {
+        LocalDateTime start = LocalDateTime.now();
         try {
             updateDevices(deviceList);
             updateSensors(sensorList);
             updateStatus(ThingStatus.ONLINE);
-        } catch (TellstickException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            logger.error("Failed to update", e);
         } catch (Exception e) {
+            logger.warn("Failed to update", e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
-            logger.error("Failed to update", e);
         }
+        monitorAdditionalRefresh(start, LocalDateTime.now());
+    }
+
+    private void monitorAdditionalRefresh(LocalDateTime start, LocalDateTime end) {
+        if (!logger.isDebugEnabled()) {
+            return;
+        }
+        long duration = ChronoUnit.MILLIS.between(start, end);
+        sumRefreshDuration += duration;
+        nbRefresh++;
+        if (duration < minRefreshDuration) {
+            minRefreshDuration = duration;
+        }
+        if (duration > maxRefreshDuration) {
+            maxRefreshDuration = duration;
+        }
+        logger.debug(
+                "{} refresh avg = {} ms min = {} max = {} (request avg = {} ms min = {} max = {}) ({} timeouts {} errors)",
+                nbRefresh, nbRefresh == 0 ? 0 : sumRefreshDuration / nbRefresh, minRefreshDuration, maxRefreshDuration,
+                controller.getAverageRequestDuration(), controller.getMinRequestDuration(),
+                controller.getMaxRequestDuration(), controller.getNbTimeouts(), controller.getNbErrors());
     }
 
     private synchronized void updateDevices(TellstickNetDevices previouslist) throws TellstickException {
@@ -258,7 +283,8 @@ public class TelldusLiveBridgeHandler extends BaseBridgeHandler implements Telld
         if (deviceStatusListener == null) {
             throw new IllegalArgumentException("It's not allowed to pass a null deviceStatusListener.");
         }
-        return deviceStatusListeners.add(deviceStatusListener);
+        return deviceStatusListeners.contains(deviceStatusListener) ? false
+                : deviceStatusListeners.add(deviceStatusListener);
     }
 
     @Override

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveDeviceController.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveDeviceController.java
@@ -14,8 +14,8 @@ package org.openhab.binding.tellstick.internal.live;
 
 import java.math.BigDecimal;
 import java.math.RoundingMode;
-import java.time.LocalDateTime;
-import java.time.temporal.ChronoUnit;
+import java.time.Duration;
+import java.time.Instant;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
@@ -83,7 +83,7 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
 
     private int nbRequest;
     private long sumRequestDuration;
-    private long minRequestDuration = 1000000;
+    private long minRequestDuration = 1_000_000;
     private long maxRequestDuration;
     private int nbTimeouts;
     private int nbErrors;
@@ -285,24 +285,29 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
     }
 
     <T> T callRestMethod(String uri, Class<T> response) throws TelldusLiveException {
-        LocalDateTime start = LocalDateTime.now();
+        Instant start = Instant.now();
         T resultObj = null;
         try {
             resultObj = innerCallRest(uri, response);
-        } catch (TimeoutException | InterruptedException | ExecutionException | JAXBException | XMLStreamException e) {
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
             logResponse(uri, e);
-            monitorAdditionalRequest(start, LocalDateTime.now(), e);
+            monitorAdditionalRequest(start, Instant.now(), e);
+            throw new TelldusLiveException(e);
+        } catch (TimeoutException | ExecutionException | JAXBException | XMLStreamException e) {
+            logResponse(uri, e);
+            monitorAdditionalRequest(start, Instant.now(), e);
             throw new TelldusLiveException(e);
         }
-        monitorAdditionalRequest(start, LocalDateTime.now(), null);
+        monitorAdditionalRequest(start, Instant.now(), null);
         return resultObj;
     }
 
-    private void monitorAdditionalRequest(LocalDateTime start, LocalDateTime end, @Nullable Throwable exception) {
+    private void monitorAdditionalRequest(Instant start, Instant end, @Nullable Throwable exception) {
         if (!logger.isDebugEnabled()) {
             return;
         }
-        long duration = ChronoUnit.MILLIS.between(start, end);
+        long duration = Duration.between(start, end).toMillis();
         sumRequestDuration += duration;
         nbRequest++;
         if (duration < minRequestDuration) {

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveDeviceController.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/TelldusLiveDeviceController.java
@@ -14,6 +14,8 @@ package org.openhab.binding.tellstick.internal.live;
 
 import java.math.BigDecimal;
 import java.math.RoundingMode;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
@@ -35,6 +37,7 @@ import org.asynchttpclient.Response;
 import org.asynchttpclient.oauth.ConsumerKey;
 import org.asynchttpclient.oauth.OAuthSignatureCalculator;
 import org.asynchttpclient.oauth.RequestToken;
+import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.tellstick.internal.TelldusBindingException;
 import org.openhab.binding.tellstick.internal.handler.TelldusDeviceController;
 import org.openhab.binding.tellstick.internal.live.xml.TelldusLiveResponse;
@@ -66,7 +69,7 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
     private final Logger logger = LoggerFactory.getLogger(TelldusLiveDeviceController.class);
     private long lastSend = 0;
     public static final long DEFAULT_INTERVAL_BETWEEN_SEND = 250;
-    static final int REQUEST_TIMEOUT_MS = 5000;
+    private static final int REQUEST_TIMEOUT_MS = 15000;
     private AsyncHttpClient client;
     static final String HTTP_API_TELLDUS_COM_XML = "http://api.telldus.com/xml/";
     static final String HTTP_TELLDUS_CLIENTS = HTTP_API_TELLDUS_COM_XML + "clients/list";
@@ -77,7 +80,13 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
     static final String HTTP_TELLDUS_DEVICE_DIM = HTTP_API_TELLDUS_COM_XML + "device/dim?id=%d&level=%d";
     static final String HTTP_TELLDUS_DEVICE_TURNOFF = HTTP_API_TELLDUS_COM_XML + "device/turnOff?id=%d";
     static final String HTTP_TELLDUS_DEVICE_TURNON = HTTP_API_TELLDUS_COM_XML + "device/turnOn?id=%d";
-    private static final int MAX_RETRIES = 3;
+
+    private int nbRequest;
+    private long sumRequestDuration;
+    private long minRequestDuration = 1000000;
+    private long maxRequestDuration;
+    private int nbTimeouts;
+    private int nbErrors;
 
     public TelldusLiveDeviceController() {
     }
@@ -87,7 +96,7 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
         try {
             client.close();
         } catch (Exception e) {
-            logger.error("Failed to close client", e);
+            logger.debug("Failed to close client", e);
         }
     }
 
@@ -101,7 +110,7 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
             Response response = client.prepareGet(HTTP_TELLDUS_CLIENTS).execute().get();
             logger.debug("Response {} statusText {}", response.getResponseBody(), response.getStatusText());
         } catch (InterruptedException | ExecutionException e) {
-            logger.error("Failed to connect", e);
+            logger.warn("Failed to connect", e);
         }
     }
 
@@ -114,7 +123,7 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
     @Override
     public void handleSendEvent(Device device, int resendCount, boolean isdimmer, Command command)
             throws TellstickException {
-        logger.info("Send {} to {}", command, device);
+        logger.debug("Send {} to {}", command, device);
         if (device instanceof TellstickNetDevice) {
             if (command == OnOffType.ON) {
                 turnOn(device);
@@ -276,37 +285,66 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
     }
 
     <T> T callRestMethod(String uri, Class<T> response) throws TelldusLiveException {
+        LocalDateTime start = LocalDateTime.now();
         T resultObj = null;
         try {
-            for (int i = 0; i < MAX_RETRIES; i++) {
-                try {
-                    resultObj = innerCallRest(uri, response);
-                    break;
-                } catch (TimeoutException e) {
-                    logger.warn("TimeoutException error in get", e);
-                } catch (InterruptedException e) {
-                    logger.warn("InterruptedException error in get", e);
-                }
-            }
-        } catch (JAXBException e) {
-            logger.warn("Encoding error in get", e);
+            resultObj = innerCallRest(uri, response);
+        } catch (TimeoutException | InterruptedException | ExecutionException | JAXBException | XMLStreamException e) {
             logResponse(uri, e);
-            throw new TelldusLiveException(e);
-        } catch (XMLStreamException e) {
-            logger.warn("Communication error in get", e);
-            logResponse(uri, e);
-            throw new TelldusLiveException(e);
-        } catch (ExecutionException e) {
-            logger.warn("ExecutionException error in get", e);
+            monitorAdditionalRequest(start, LocalDateTime.now(), e);
             throw new TelldusLiveException(e);
         }
+        monitorAdditionalRequest(start, LocalDateTime.now(), null);
         return resultObj;
+    }
+
+    private void monitorAdditionalRequest(LocalDateTime start, LocalDateTime end, @Nullable Throwable exception) {
+        if (!logger.isDebugEnabled()) {
+            return;
+        }
+        long duration = ChronoUnit.MILLIS.between(start, end);
+        sumRequestDuration += duration;
+        nbRequest++;
+        if (duration < minRequestDuration) {
+            minRequestDuration = duration;
+        }
+        if (duration > maxRequestDuration) {
+            maxRequestDuration = duration;
+        }
+        if (exception instanceof TimeoutException) {
+            nbTimeouts++;
+        } else if (exception != null) {
+            nbErrors++;
+        }
+    }
+
+    public long getAverageRequestDuration() {
+        return nbRequest == 0 ? 0 : sumRequestDuration / nbRequest;
+    }
+
+    public long getMinRequestDuration() {
+        return minRequestDuration;
+    }
+
+    public long getMaxRequestDuration() {
+        return maxRequestDuration;
+    }
+
+    public int getNbTimeouts() {
+        return nbTimeouts;
+    }
+
+    public int getNbErrors() {
+        return nbErrors;
     }
 
     private <T> T innerCallRest(String uri, Class<T> response) throws InterruptedException, ExecutionException,
             TimeoutException, JAXBException, FactoryConfigurationError, XMLStreamException {
         Future<Response> future = client.prepareGet(uri).execute();
         Response resp = future.get(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+        if (resp.getStatusCode() != 200) {
+            throw new ExecutionException("HTTP request returned status code " + resp.getStatusCode(), null);
+        }
         // TelldusLiveHandler.logger.info("Devices" + resp.getResponseBody());
         JAXBContext jc = JAXBContext.newInstance(response);
         XMLInputFactory xif = XMLInputFactory.newInstance();
@@ -324,8 +362,6 @@ public class TelldusLiveDeviceController implements DeviceChangeListener, Sensor
     }
 
     private void logResponse(String uri, Exception e) {
-        if (e != null) {
-            logger.warn("Request [{}] Failure:{}", uri, e.getMessage());
-        }
+        logger.warn("Request [{}] failed: {} {}", uri, e.getClass().getSimpleName(), e.getMessage());
     }
 }

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/xml/TellstickNetDevice.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/xml/TellstickNetDevice.java
@@ -151,10 +151,7 @@ public class TellstickNetDevice implements Device {
             return false;
         }
         TellstickNetDevice other = (TellstickNetDevice) obj;
-        if (deviceId != other.deviceId) {
-            return false;
-        }
-        return true;
+        return deviceId == other.deviceId;
     }
 
     public int getMethods() {

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/xml/TellstickNetSensor.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/live/xml/TellstickNetSensor.java
@@ -143,10 +143,7 @@ public class TellstickNetSensor implements Device {
             return false;
         }
         TellstickNetSensor other = (TellstickNetSensor) obj;
-        if (deviceId != other.deviceId) {
-            return false;
-        }
-        return true;
+        return deviceId == other.deviceId;
     }
 
     public void setUpdated(boolean b) {

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalBridgeHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalBridgeHandler.java
@@ -13,9 +13,9 @@
 package org.openhab.binding.tellstick.internal.local;
 
 import java.time.Duration;
-import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ScheduledFuture;
 import java.util.concurrent.TimeUnit;
 
@@ -59,7 +59,7 @@ public class TelldusLocalBridgeHandler extends BaseBridgeHandler implements Tell
     private TellstickLocalDevicesDTO deviceList = null;
     private TellstickLocalSensorsDTO sensorList = null;
     private TelldusLocalDeviceController controller = null;
-    private List<DeviceStatusListener> deviceStatusListeners = Collections.synchronizedList(new ArrayList<>());
+    private Set<DeviceStatusListener> deviceStatusListeners = ConcurrentHashMap.newKeySet();
     private final HttpClient httpClient;
     private ScheduledFuture<?> pollingJob;
     /**
@@ -231,8 +231,7 @@ public class TelldusLocalBridgeHandler extends BaseBridgeHandler implements Tell
         if (deviceStatusListener == null) {
             throw new IllegalArgumentException("It's not allowed to pass a null deviceStatusListener.");
         }
-        return deviceStatusListeners.contains(deviceStatusListener) ? false
-                : deviceStatusListeners.add(deviceStatusListener);
+        return deviceStatusListeners.add(deviceStatusListener);
     }
 
     @Override

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalBridgeHandler.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalBridgeHandler.java
@@ -50,7 +50,7 @@ import org.tellstick.device.iface.Device;
  * to the framework. All {@link TelldusDevicesHandler}s use the
  * {@link TelldusLocalDeviceController} to execute the actual commands.
  *
- * @author Jan Gustafsson- Initial contribution
+ * @author Jan Gustafsson - Initial contribution
  */
 public class TelldusLocalBridgeHandler extends BaseBridgeHandler implements TelldusBridgeHandler {
 
@@ -231,7 +231,8 @@ public class TelldusLocalBridgeHandler extends BaseBridgeHandler implements Tell
         if (deviceStatusListener == null) {
             throw new IllegalArgumentException("It's not allowed to pass a null deviceStatusListener.");
         }
-        return deviceStatusListeners.add(deviceStatusListener);
+        return deviceStatusListeners.contains(deviceStatusListener) ? false
+                : deviceStatusListeners.add(deviceStatusListener);
     }
 
     @Override

--- a/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalDeviceController.java
+++ b/bundles/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/local/TelldusLocalDeviceController.java
@@ -56,7 +56,6 @@ public class TelldusLocalDeviceController implements DeviceChangeListener, Senso
     private final Logger logger = LoggerFactory.getLogger(TelldusLocalDeviceController.class);
     private long lastSend = 0;
     public static final long DEFAULT_INTERVAL_BETWEEN_SEND_SEC = 250;
-    static final int REQUEST_TIMEOUT_MS = 5000;
     private final HttpClient httpClient;
     private final Gson gson = new Gson();
     private String localApiUrl;

